src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs

```diff
@@ -0,0 +1,78 @@
+// <copyright file="HttpInMetricsListener.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.AspNetCore.Http;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Instrumentation.AspNetCore.Implementation
+{
+    internal class HttpInMetricsListener : ListenerHandler
+    {
+        private readonly PropertyFetcher<HttpContext> stopContextFetcher = new PropertyFetcher<HttpContext>("HttpContext");
+        private readonly Meter meter;
+
+        private Counter<long> httpServerRequestCount;
+
+        public HttpInMetricsListener(string name, Meter meter)
+            : base(name)
+        {
+            this.meter = meter;
+
+            // TODO:
+            //   In the future, this instrumentation should produce the http.server.duration metric which will likely be represented as a histogram.
+            //   See: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#http-server
+            //
+            //   Histograms are not yet supported by the SDK.
+            //
+            //   For now we produce a count metric called http.server.request_count just for demonstration purposes.
+            //   This metric is not defined by the in the semantic conventions.
+            this.httpServerRequestCount = meter.CreateCounter<long>("http.server.request_count", null, "The number of HTTP requests processed.");
+        }
+
+        public override void OnStopActivity(Activity activity, object payload)
+        {
+            HttpContext context = this.stopContextFetcher.Fetch(payload);
+            if (context == null)
+            {
+                AspNetCoreInstrumentationEventSource.Log.NullPayload(nameof(HttpInMetricsListener), nameof(this.OnStopActivity));
+                return;
+            }
+
+            // TODO: Prometheus pulls metrics by invoking the /metrics endpoint. Decide if it makes sense to suppress this.
+            // Below is just a temporary way of achieving this suppression for metrics (we should consider suppressing traces too).
```

Comment:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/metrics/src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterMetricsHttpServer.cs#L114
Traces would be suppressed, *similar* to this.

Reply:
Ah yea the comment could probably be more clear - this comment was left over from my original PR from awhile back.

I meant suppressing invocations of the `/metrics` endpoint exposed via the ASP.NET Core middleware - pretty sure this would cause an activity to be generated. At least that's what I recall when running this with the old metrics code and exporting to prometheus, though I haven't tried out prometheus now that we've resurrected all the code.

Reply:
we haven't resurrected the middleware version of Prometheus. The in-proc server using HttpListener is resurrected, and it already does suppress. (but since there is no instrumentation for it, it shouldn't get captured anyway).
We can resurrect the asp.net core middleware based one and add the suppress flag there, so activity wont be recorded from that operation.